core/module_protocols/core_ssh.py

```diff
@@ -20,14 +20,18 @@ def ssh_brute_force(host, ports, usernames, passwords, timeout):
         paramiko_logger.disabled = True
         ssh = paramiko.SSHClient()
         ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-        ssh.connect(
-            hostname=host,
-            username=usernames,
-            password=passwords,
-            port=int(ports),
-            timeout=int(timeout)
-        )
-        ssh.close()
+        try:
+            ssh.connect(
+                hostname=host,
+                username=usernames,
+                password=passwords,
+                port=int(ports),
+                timeout=int(timeout)
+            )
+            ssh.close()
+        except paramiko.ssh_exception.AuthenticationException as e:
+            if not passwords:
+                ssh.get_transport().auth_none(usernames)
```

Comment:
Consider more readable version without try/except block.
Please also note the connection doesn't get closed for the empty password case in your code.
```suggestion

        if passwords:
            ssh.connect(
                hostname=host,
                username=usernames,
                password=passwords,
                port=int(ports),
                timeout=int(timeout),
            )
        else:
            ssh.get_transport().auth_none(usernames)

        ssh.close()

```

Reply:
@arkid15r, kindly review the new commit i have made following your suggestions.